torch/fx/experimental/proxy_tensor.py

```diff
@@ -59,8 +60,12 @@ def wrap_with_proxy(e, proxy):
     else:
         return real_out
 
+from torch.testing._internal.composite_compliance import check_metadata_consistency
 
 def proxy_call(func_overload, args, kwargs=None):
+    if func_overload == torch.ops.prim.device.default:
+        return args[0].fake_device
```

Comment:
Don't you need to do the `in_kernel_dispatch` thing here ? `args[0].device` should correctly return the `fake_device` or `meta` depending on the context